xdsl/dialects/vector.py

```diff
@@ -54,4 +54,99 @@ def get(vector: Operation | SSAValue, ref: Operation | SSAValue,
+                               ])
+
+
+@irdl_op_definition
+class FMA(Operation):
+    name = "vector.fma"
+    lhs: Annotated[Operand, VectorType]
+    rhs: Annotated[Operand, VectorType]
+    acc: Annotated[Operand, VectorType]
+    res: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        res_type = self.res.typ.element_type
+        res_shape = self.res.typ.get_shape()
+
+        if res_type != self.lhs.typ.element_type or res_type != self.rhs.typ.element_type or res_type != self.acc.typ.element_type:
+            raise VerifyException(
+                "Result vector type must match with all source vectors.")
+
+        if res_shape != self.lhs.typ.get_shape(
+        ) or res_shape != self.rhs.typ.get_shape(
+        ) or res_shape != self.acc.typ.get_shape():
+            raise VerifyException(
+                "Result vector shape must match with all source vector shapes."
+            )
+
+    @staticmethod
+    def get(lhs: Operation | SSAValue, rhs: Operation | SSAValue,
+            acc: Operation | SSAValue) -> FMA:
+        return FMA.build(operands=[lhs, rhs, acc],
+                         result_types=[
+                             VectorType.from_type_and_list(
+                                 SSAValue.get(lhs).typ.element_type)
+                         ])
+
+
+@irdl_op_definition
+class Maskedload(Operation):
+    name = "vector.maskedload"
+    memref: Annotated[Operand, MemRefType]
+    indices: Annotated[VarOperand, IndexType]
+    mask: Annotated[Operand, VectorType]
+    passthrough: Annotated[Operand, VectorType]
+    res: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        memref_type = self.memref.typ.element_type
+
+        if memref_type != self.res.typ.element_type or memref_type != self.passthrough.typ.element_type:
+            raise VerifyException(
+                "MemRef element type should match the result vector and passthrough vector element type."
+            )
+
+        if len(self.res.typ.get_shape()) != 1:
+            raise VerifyException("Expected a rank 1 result vector.")
+
+        if len(self.mask.typ.get_shape()) != 1:
+            raise VerifyException("Expected a rank 1 mask vector.")
+
+        if self.mask.typ.element_type != i1:
+            raise VerifyException("Expected mask element type to be i1.")
+
+        if self.memref.typ.get_num_dims() != len(self.indices):
+            raise VerifyException(
+                "Expected an index for each memref dimension.")
+
+    @staticmethod
+    def get(memref: SSAValue | Operation, indices: List[SSAValue | Operation],
+            mask: SSAValue | Operation,
+            passthrough: SSAValue | Operation) -> Maskedload:
+        return Maskedload.build(operands=[memref, indices, mask, passthrough],
+                                result_types=[
+                                    VectorType.from_type_and_list(
+                                        SSAValue.get(memref).typ.element_type)
```

Comment:
I am confused. Isn't the result always a 1-D vector? The memref that gets passed into this can be multidimensional, right?

This comes just from me reading through the docu, I am not extremely well educated in the space :)

Reply:
> Isn't the result always a 1-D vector? The memref that gets passed into this can be multidimensional, right?

Yes

Reply:
Yes, I understand that. But it seems to me that this constructor could fail in certain cases. If you were try to construct this exact operation you mention using this constructor, I feel like the type of the result vector will be wrong as this just copies the type of the memref, while it should copy the type of the passthrough, I think.

Reply:
The element type of memref, result vector and passthrough vector is supposed to be exactly same. AFAIU, we are specifying the result vector with element type same as memref without actually using memref shape anywhere.

Reply:
>  I feel like this constructor needs a different name then, that looks quite confusing to me...

I agree. 

How about `VectorType.from_type_and_shape()` ?

Reply:
Could you remove the default argument for the shape in this constructor? I discussed this quite a bit with Mathieu and we agreed that this is the way to go.

Also, please rename the constructor to `from_element_type_and_shape`.

Reply:
I'll do this in a separate PR.